### ISSUE TITLE
Investigate if keeping icache enabled is okay

### ISSUE THREAD

**Issue by gamozolabs** [gamozolabs/applepie#18]

**Comment by gamozolabs** [comment]
Cannot enable as we do not have the hooks requires for self-modifying code when running hypervisor code. We could maybe reset icache entries based on dirtied physical memory during a VM entry, but this is probably more expensive than it's worth.

We could also flush icache entries after hypervisor exits (allowing the icache to be used during a single bochs step_cpu() invocation), however this flushing is more expensive than just disabling the caches.

Maybe by changing the cache size we can find a middle ground, but it's not something we can just enable out of the box.

**Comment by gamozolabs** [comment]
Okay we improved this by making a `hypervisor_context_switches` global which tracks number of context switches from the hypervisor. We tag icache entries with this number so we can lazily invalidate cache entries as they're used if they're older than the most recent hypervisor context switch.

TLDR: We only use icache entries from the current context switch
